tests/test_utils.py

```diff
@@ -0,0 +1,179 @@
+        if oa.nargs == 0 and oa.const is True and oa.default is False:
+            default_false_values.update(**{oa.dest: False})
+    return default_false_values
+
+
+def get_gather_args_with_mandatory_values():
+    # Get these from the parser rather than having to keep a manual list.
+    parser = subutils.build_gather_options_parser([])
+    optional_actions = parser._get_optional_actions()
+    mandatory_value_args = []
+    for oa in optional_actions:
+        if oa.nargs in ("?", "+", 1):
+            mandatory_value_args.append(oa.dest)
+    return mandatory_value_args
+
+
+gather_default_false_values = get_gather_default_false_values()
+gather_args_with_mandatory_values = get_gather_args_with_mandatory_values()
+
+
+@pytest.mark.parametrize("args,expected", [
+    (
+        "gather dap --dap=someurl --suffix=.gov",
+        {
+            "_": ["dap"],
+            "dap": "someurl",
+            "suffix": ".gov",
+            **gather_default_false_values,
+        }
+    ),
+    (
+        "".join([
+            "gather dap --suffix=.gov --dap=",
+            "https://analytics.usa.gov/data/live/sites-extended.csv"]),
+        {
+            "_": ["dap"],
+            **gather_default_false_values,
+            "suffix": ".gov",
+            "dap": "https://analytics.usa.gov/data/live/sites-extended.csv",
+        }
+    ),
+    (
+        "".join([
+            "./gather censys,dap,private --suffix=.gov --dap=",
+            "https://analytics.usa.gov/data/live/sites-extended.csv",
+            " --private=/path/to/private-research.csv --parents=",
+            "https://github.com/GSA/data/raw/master/dotgov-domains/current-federal.csv",
+        ]),
+        {
+            '_': ['censys,dap,private'],
+            **gather_default_false_values,
+            'suffix': '.gov',
+            'dap': 'https://analytics.usa.gov/data/live/sites-extended.csv',
+            'private': '/path/to/private-research.csv',
+            'parents': 'https://github.com/GSA/data/raw/master/dotgov-domains/current-federal.csv',
+        }
+    ),
+    (
+        "".join([
+            "./gather censys,dap,private --suffix=.gov --dap=",
+            "https://analytics.usa.gov/data/live/sites-extended.csv",
+            " --private=/path/to/private-research.csv --parents=",
+            "https://github.com/GSA/data/raw/master/dotgov-domains/current-federal.csv",
+        ]),
+        {
+            '_': ['censys,dap,private'],
+            **gather_default_false_values,
+            'suffix': '.gov',
+            'dap': 'https://analytics.usa.gov/data/live/sites-extended.csv',
+            'private': '/path/to/private-research.csv',
+            'parents': 'https://github.com/GSA/data/raw/master/dotgov-domains/current-federal.csv',
+        }
+    ),
+    (
+        "".join([
+            "./gather dap --suffix=.gov --dap=",
+            "https://analytics.usa.gov/data/live/sites-extended.csv",
+            " --ignore-www",
```

Comment:
Maybe I'm not intuiting what the change means -- do `--ignore-www` and `-ignore_www` both work now?

Reply:
On the command line only `--ignore-www` will work; `ignore_www` is what the property name will be after argparse processes them.